jhu-idc/idc-isle-dc
# Integrates idc_defaults module into iDC and adds constraint on member_of field

This integrates the `idc_defaults` module into iDC. The module adds a constraint for the `field_member_of` and `field_access_terms` on repository items.  It will allow those fields to be set based on the access the user has to certain content.  It will complain and throw validation exceptions if the user tries to set those fields (via UI or ingest) to anything they do not have access to. 

Also starts some role-based testing infrastructure for it.  Only starts the infrastructure - it needs to be fleshed out more once "validate:true" is turned on in migrations. 

To test the new `field_member_of` constraint:
1) `make reset` on this branch
2) run the 21-role-permissions-tests manually (`make test test=21-role-permission-tests`). They will do some ingests and create a CLA admin. 
3) log in as `claAdmin` (password: `password`) - this admin was created via the test suite
4) As this admin, create a new repository item and try to put it in `Collection C (Zoo Animals)`.  You will have to fill in the `field_member_of`  with `Collection C (Zoo Animals)` and click on submit. Then it should fail with an error at the top of the page telling you that you can't ingest into that collection. 

The other fix involves setting the `field_access_terms` to the wrong thing. In the UI, it will only offer values that the user has access to.  We need to test this via ingest, but we can't test that until `validate: true` is enabled on ingest.
